shell/server/admin/system-status-server.js

```diff
@@ -1,9 +1,61 @@
-const Crypto = Npm.require("crypto");
+import { Meteor } from "meteor/meteor";
+import { Router } from "meteor/iron:router";
+import { Random } from "meteor/random";
+import Crypto from "crypto";
+import Fs from "fs";
 
 const hashSessionId = function (sessionId) {
   return Crypto.createHash("sha256").update(sessionId).digest("base64");
 };
 
+// An in-memory map of random id -> timestamp, used to authorize requests to download the system
+// log.
+const SYSTEM_LOG_DOWNLOAD_TOKENS = {};
+
+Meteor.methods({
+  adminGetServerLogDownloadToken() {
+    const db = this.connection.sandstormDb;
+    if (!db.isAdminById(this.userId)) {
+      throw new Meteor.Error(403, "User must be admin to download system log.");
+    }
+
+    const token = Random.id();
+    SYSTEM_LOG_DOWNLOAD_TOKENS[token] = Date.now();
+    return token;
+  },
+});
+
+Router.map(function () {
+  this.route("adminDownloadServerLog", {
+    where: "server",
+    path: "/admin/status/server-log/:tokenId",
+    action() {
+      const token = this.params.tokenId;
+      const response = this.response;
+      const issueDate = SYSTEM_LOG_DOWNLOAD_TOKENS[token];
+      SYSTEM_LOG_DOWNLOAD_TOKENS[token] = undefined; // Clear the token from the in-memory map.
```

Comment:
Yeah, it's tiny, requires an admin to take an action to trigger, goes away on frontend restart, and so on.

I guess I could `setTimeout()` a function to remove the newly-added entry from the map after, say, ~2x the valid duration, though, which wouldn't be much of a hassle.